ecosystem/sep-0043.md

```diff
@@ -0,0 +1,146 @@
+may surface:
+
+```typescript
+{
+  unhandled: { // internal wallet error, likely caused by the wallet logic itself
+    message: 'The wallet encountered an internal error. Please try again or contact the wallet if the problem persists.',
+    code: -1
+  },
+  external: { // external service (Horizon, RPC, etc.) returned an error
+    message: 'An error occurred with Horizon. Please try again.',
+    code: -2,
+    ext: ['Operation 2: Invalid "amount" value.', "Operation 5: Asset issuer is required."] // malformed tx error example
+  },
+  invalid: { // client app request is invalid (wrong parameters, invalid transaction XDR, etc.)
+    message: 'Request is invalid. Please check the details and try again.',
+    code: -3,
+    ext: ['Invalid transaction XDR'] // example error
+  },
+  rejected: { // user rejected the request, client app should not try to retry the request
+    message: 'The user rejected this request.',
+    code: -4
+  }
+}
+
+```
+
+### Methods
+
+### getAddress
+
+`getAddress` will provide the public key the wallet is signing for. This may be the account loaded in the wallet, or it
+may be a different account that the wallet is the signer for.
+
+### signTransaction
+
+`signTransaction` will take an XDR provided by the requester and apply a signature to it before returning a signed XDR
+and the signer address back to the requester. The first parameter is the XDR string while the second parameter is a set
+of options. Options allow a user to specify what network they intend the transaction to be signed on. They can do so by
+passing the network's passphrase as `networkPassphrase`. Options also allow a wallet to specify which account they're
+seeking a signature for as `address`. If a wallet holds numerous addresses, it can use this param to ensure it is
+signing with the intended address. Options also allow a user to set a `submit` flag, which will allow a user to sign a
+transaction and immediately submit it to a network. If a user would like to configure the URL of the network to submit
+to, they can do so by setting `submitUrl`.
+
+### signAuthEntry
+
+Similar to its transaction counterpart, `signAuthEntry` will take an XDR provided by the requester and apply a signature
+to it before returning a signature and the signer address back to the requester. The first parameter is the XDR string
+while the second parameter is a set of options. Options allow a user to specify what network they intend the transaction
+to be signed on. They can do so by passing the network's passphrase as `networkPassphrase`. Options also allow a wallet
+to specify which account they're seeking a signature for as `address`. If a wallet holds numerous addresses, it can use
+this param to ensure it is signing with the intended address.
+
+### signMessage
+
+Similar to its transaction and aither entry counterparts, `signMessage` will take an arbitrary message provided by the
```

Comment:
typo:
```suggestion
Similar to its transaction and auth entry counterparts, `signMessage` will take an arbitrary message provided by the
```

also, wouldn't it be `Contrary` rather than `Similar` given that it's now arbitrary rather than well-defined or structured data?

Reply:
I was trying to convey that all 3 are similar in that you're sending _something_ to the wallet, it signs, and then returns the signed version. But I can make that more clear